### PR TITLE
Feature/sdl 0262 New vehicle data SeatOccupancy

### PR DESCRIPTION
**Pull request by KhrystynaDubovyk**
Guidelines updates according to the proposal [SDL-0262 New vehicle data SeatOccupancy](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0262-New-vehicle-data-SeatOccupancy.md) 

This PR is **ready** for review.

### Summary
Expand `VehicleDataType` enum with new param
Add new struct `SeatStatus`
Add new struct `SeatOccupancy`
Add `seatOccupancy` to SubscribeVehicleData, UnsubscribeVehicleData, GetVehicleData, OnVehicleData RPCs

### CODE REVIEW

**Comment by KhrystynaDubovyk** [comment]
@atiwari9 please review this PR

**Review by atiwari9 — APPROVED** [review]

**Comment by LitvinenkoIra** [comment]
@JackLivio This PR is ready for Livio review. Thank you!

File: docs/VehicleInfo/SubscribeVehicleData/index.md [hunk 1]
```diff
@@ -103,6 +104,7 @@ If multiple applications are trying to restore the same subscription, SDL should
 |stabilityControlsStatus|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
 |windowStatus|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
 |handsOffSteering|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
+|seatOccupancy|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false|
```
Comment (jacobkeeler) [inline docs/VehicleInfo/SubscribeVehicleData/index.md:107]:
```suggestion
|seatOccupancy|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
```

File: docs/VehicleInfo/UnsubscribeVehicleData/index.md [hunk 2]
```diff
@@ -103,6 +104,7 @@ In the case during data resumption with multiple applications, a subscription is
 |stabilityControlsStatus|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
 |windowStatus|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
 |handsOffSteering|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
+|seatOccupancy|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false|
```
Comment (jacobkeeler) [inline docs/VehicleInfo/UnsubscribeVehicleData/index.md:107]:
```suggestion
|seatOccupancy|[Common.VehicleDataResult](../../common/structs/#vehicledataresult)|false||
```

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

**Review by jacobkeeler — APPROVED** [review]

**Comment by jacobkeeler** [comment]
@LitvinenkoIra please fix conflicts

**Pull request merged**